### DISCUSSION TITLE
Undefined variable $attributes

### DISCUSSION THREAD

**Discussion by cgunnels** [livewire/livewire#8205] [1. Help] [answered]
Using Livewire 3. 
In my parent blade file I have: `<livewire:state-select class="shadow appearance-none ..." />` but then laravel is throwing the undefined $attributes variable error. In all the other components there is the syntax `{{ $attributes->merge(['class' => '...']) }}` but mine isn't working. What am I missing?

My StateSelect.php is
```
<?php

namespace App\Livewire;

use Livewire\Component;

class StateSelect extends Component
{
    public array $states = [];

    public function mount()
    {
        $this->states = config('states.list');
    }

    public function render()
    {
        return view('livewire.state-select');
    }
}
```

My livewire blade:
```
<div {{ $attributes->merge(['class' => 'mt-4']) }}>

    <x-input-label for="state"/>

    <select id="state" class="block mt-1 w-full" name="state" required>

        @foreach ($states as $state)

            <option value="{{ $state }}" {{ old('state') == $state ? 'selected' : '' }}>{{ $state }}</option>

        @endforeach

    </select>

    <x-input-error :messages="$errors->get('state')" class="mt-2"/>

</div>
```

**Comment by joshhanley** [comment]
@cgunnels Livewire doesn't have an $attributes bag, that's only for Blade components 🙂

**Comment by cgunnels** [comment]
Gotcha, is there a way to pass attributes? I want to reuse the state select component in multiple locations with different styling.

**Reply by joshhanley** [reply, marked as answer]
The only way to do it is manually, like make a $class property on the component or something like that. Because otherwise the component won't know about the passed in data on subsequent renders.
